src/main/java/com/minecolonies/rankup/util/AccountingUtils.java

```diff
@@ -0,0 +1,524 @@
+        }
+        return sql.getDataSource(jdbcUrl);
+    }
+
+    private String getURI()
+    {
+        if (getDatabasesConfig().database.equalsIgnoreCase("mysql"))
+        {
+            return "jdbc:mysql://"
+                     + getDatabasesConfig().sqlUsername
+                     + ":"
+                     + getDatabasesConfig().sqlPassword
+                     + "@"
+                     + getDatabasesConfig().sqlAddress
+                     + "/"
+                     + getDatabasesConfig().sqlDatabase;
+        }
+        return "jdbc:h2:" + plugin.getConfigDir() + "/h2/playerstats";
+    }
+
+    private Connection getConn()
+    {
+        if (conn == null)
+        {
+            String uri = getURI();
+            try
+            {
+                conn = getDataSource(uri).getConnection();
+            }
+            catch (SQLException e)
+            {
+                e.printStackTrace();
+            }
+        }
+        return conn;
+    }
+
+    private ResultSet getQuery(final String query)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                final Connection conn = getConn();
+
+                if (conn != null)
+                {
+                    return conn.createStatement().executeQuery(query);
+                }
+            }
+        }
+        catch (NoModuleException | SQLException e)
+        {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private String toDate(final String date)
+    {
+        final String newDate = CommonUtils.dateFormat(date);
+        if (getDatabasesConfig().database.equalsIgnoreCase("h2"))
+        {
+            return "to_date('" + newDate + "','yyyy-mm-dd')";
+        }
+        return "STR_TO_DATE('" + newDate + "','%Y-%m-%d')";
+    }
+
+    public void createTableIfNeeded()
+    {
+        try
+        {
+            Statement stmt = getConn().createStatement();
+
+            stmt.execute("CREATE TABLE IF NOT EXISTS " + table_id + "("
+                           + UUID_COLUMN + " varchar(255) NOT NULL, "
+                           + PLAYER_NAME_COLUMN + " varchar(255) NOT NULL, "
+                           + JOIN_DATE_COLUMN + " DATE NOT NULL, "
+                           + LAST_JOIN_COLUMN + " DATE NOT NULL, "
+                           + TIME_PLAYED_COLUMN + " int NOT NULL, "
+                           + "PRIMARY KEY(" + UUID_COLUMN + ") )");
+        }
+        catch (SQLException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    public String getPlayerName(final UUID uuid)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                final ResultSet results = getQuery("SELECT " + PLAYER_NAME_COLUMN + " FROM " + table_id + " WHERE " + UUID_COLUMN + " = '" + uuid + "'");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal " WHERE " 7 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal " FROM " 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "SELECT " 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)